SabreTools.Models/PlayStation4/AppPkgHeader.cs

```diff
@@ -0,0 +1,238 @@
+
+namespace SabreTools.Models.PlayStation4
+{
+    /// <see href="https://www.psdevwiki.com/ps4/PKG_files"/>
+    [StructLayout(LayoutKind.Sequential)]
+    public class AppPkgHeader
+    {
+        /// <summary>
+        /// 0x7F434E54
+        /// ".CNT"
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
+        public byte[]? Magic;
+
+        /// <summary>
+        /// PKG Type
+        /// </summary>
+        public uint Type;
+
+        /// <summary>
+        /// PKG Unknown Field
+        /// </summary>
+        public uint PKGUnknown;
+
+        /// <summary>
+        /// PKG File count
+        /// </summary>
+        public uint FileCount;
+
+        /// <summary>
+        /// PKG Entry count
+        /// </summary>
+        public uint EntryCount;
+
+        /// <summary>
+        /// SC Entry count
+        /// </summary>
+        public ushort SCEntryCount;
+
+        /// <summary>
+        /// PKG Entry count (duplicated)
+        /// </summary>
+        public ushort EntryCount2;
+
+        /// <summary>
+        /// PKG Table offset
+        /// </summary>
+        public uint TableOffset;
+
+        /// <summary>
+        /// PKG Entry data size
+        /// </summary>
+        public uint EntryDataSize;
+
+        /// <summary>
+        /// PKG Body offset
+        /// </summary>
+        public ulong BodyOffset;
+
+        /// <summary>
+        /// PKG Body size
+        /// </summary>
+        public ulong BodySize;
+
+        /// <summary>
+        /// PKG Content offset
+        /// </summary>
+        public ulong ContentOffset;
+
+        /// <summary>
+        /// PKG Content size
+        /// </summary>
+        public ulong ContentSize;
+
+        /// <summary>
+        /// PKG Content ID
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 36)]
+        public string? ContentID;
+        
+        /// <summary>
+        /// PKG Content Padding (Zeroes)
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 12)]
+        public byte[]? ContentZeroes;
+
+        /// <summary>
+        /// PKG DRM Type
+        /// </summary>
+        public uint DRMType;
+
+        /// <summary>
+        /// PKG Content Type
+        /// </summary>
+        public uint ContentType;
+
+        /// <summary>
+        /// PKG Content Flags
+        /// </summary>
+        public uint ContentFlags;
+
+        /// <summary>
+        /// PKG Promote Size
+        /// </summary>
+        public uint PromoteSize;
+
+        /// <summary>
+        /// PKG Version Date
+        /// </summary>
+        public uint VersionDate;
+
+        /// <summary>
+        /// PKG Content Flags
+        /// </summary>
+        public uint VersionHash;
+        
+        /// <summary>
+        /// PKG Padding Section 1
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 120)]
+        public byte[]? Zeroes1;

```

Comment:
I might be missing something, but `pkg_0x088` through `pkg_0x094` only seem to make up 16 bytes of unknown data, not 120 bytes.

Also it seems like `pkg_drm_type_version` is missing as well.

Reply:
You are reading it wrong, there is a gap they don't explicitly include.
Also, `pkg_drm_type_version` is just zeroes on the discs I have.

Reply:
That's unfortunate as I could only really go off of the linked wiki page. I'll trust that this is the case. then.